lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -548,22 +551,22 @@ private void tryLocks(
     }
 
     @VisibleForTesting
-    void logSlowLockAcquisition(String lockId, LockClient currentHolder, long durationMillis) {
+    void logSlowLockAcquisition(LockDescriptor lockDescriptor, LockClient currentHolder, long durationMillis) {
         final String slowLockLogMessage = "Blocked for {} ms to acquire lock {} {}.";
+        final String lockId = lockDescriptor.getLockIdAsString();
 
-        // Note: The construction of params is pushed into the branches, as it may be expensive.
+        List<Arg<?>> logArgs = new ArrayList<Arg<?>>();
```

Comment:
Explicitly note the comment that you removed!!! The code was duplicated given that there is an overhead here that we don't want to pay even when we're not using it! You can always do some kind of `UnsafeArg.of("lockHoldersIfEnabled", Optional<>)` rather than a list of log args here.